data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/NestedSyntaxConverterUtil.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.logstash.mapping;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+class NestedSyntaxConverterUtil {
+    private NestedSyntaxConverterUtil() {}
+
+    public static Object checkAndConvertLogstashNestedSyntax(Object logstashAttributeValue) {
+        if (logstashAttributeValue instanceof String && ((String) logstashAttributeValue).startsWith("[") && ((String) logstashAttributeValue).endsWith("]")) {
+            return convertAttributeValueNestedSyntax(logstashAttributeValue);
+        }
+        return logstashAttributeValue;
+    }
+
+    private static String convertAttributeValueNestedSyntax(Object attributeValue) {
+        StringBuilder convertedAttribute = new StringBuilder();
+        String nestedSyntaxRegex = "\\[([^\\]]+)\\]";
```

Comment:
Consider making them static constant.

Reply:
This regex has some flaws in it. For example, I things like `[foo[bar]` will pass, among others that are invalid. I think this pattern should work to allow one or more non-empty `[]` that does not contain brackets inside of it.  

```
(\\[([^\\]\\[]+)\\])+
```

Reply:
The easiest way to do use this pattern would be something like this. I am sure there is a way to go to each `[foo]` section and append a `/` like you are doing, but this way may be easier

```
Matcher matcher = NESTED_SYNTAX_PATTERN.matcher(logstashString);

if (matcher.matches() {
     String result =  logstatshString.replaceAll("\\]\\[", "/").replaceAll("\\[|\\]", "/");
     return result;
}
```

Reply:
Thanks for the feedback Taylor. I'll work on regex. It's very hard to find the regex for complete nested syntax string. That's the reason I'm looking for groups of []. One way to do it is just check starting and ending characters like`^\[.+\]$`.

I also wrote the regex based on the assumption that we get valid Logstash configuration.